<div class="col-xs-12 col-sm-9 col-md-8 main-content">
 <section class="multi-line-signature">
  <span class="returntype">
  </span>
  <span class="name ">
   lastWhere
  </span>
  (
  <wbr>
  <span class="parameter" id="lastWhere-param-test">
   <span class="type-annotation">
    bool
   </span>
   <span class="parameter-name">
    test
   </span>
   (
   <span class="parameter" id="test-param-element">
    <span class="type-annotation">
     T
    </span>
    <span class="parameter-name">
     element
    </span>
   </span>
   )
  </span>
  , {
  <span class="parameter" id="lastWhere-param-defaultValue">
   <span class="type-annotation">
    Object
   </span>
   <span class="parameter-name">
    defaultValue
   </span>
   ()
  </span>
  })
 </section>
 <section class="desc markdown">
  <p>
   Finds the last element in this stream matching
   <code>
    test
   </code>
   .
  </p>
  <p>
   As
   <a class="" href="dart-async/Stream/firstWhere.html">
    firstWhere
   </a>
   , except that the last matching element is found.
That means that the result cannot be provided before this stream
is done.
  </p>
 </section>
 <section class="summary source-code" id="source">
  <h2>
   <span>
    Source
   </span>
  </h2>
  <pre>
   <code class="prettyprint lang-dart">
    Future&lt;dynamic&gt; lastWhere(bool test(T element), {Object defaultValue()}) {
  _Future&lt;dynamic&gt; future = new _Future();
  T result = null;
  bool foundResult = false;
  StreamSubscription subscription;
  subscription = this.listen(
    (T value) {
      _runUserCode(
        () =&gt; true == test(value),
        (bool isMatch) {
          if (isMatch) {
            foundResult = true;
            result = value;
          }
        },
        _cancelAndErrorClosure(subscription, future)
      );
    },
    onError: future._completeError,
    onDone: () {
      if (foundResult) {
        future._complete(result);
        return;
      }
      if (defaultValue != null) {
        _runUserCode(defaultValue, future._complete, future._completeError);
        return;
      }
      try {
        throw IterableElementError.noElement();
      } catch (e, s) {
        _completeWithErrorCallback(future, e, s);
      }
    },
    cancelOnError: true);
  return future;
}
   </code>
  </pre>
 </section>
</div>
